Fix use of `_additional_code` in for loops (#2040)
pyccel/pyccel: CHANGELOG.md

-   #1972 : Simplified `printf` statement for Literal String.
-   #2026 : Fix missing loop in slice assignment.
-   #2008 : Ensure list/set/dict assignment is recognised as a reference.

> -   #2039 : Ensure expressions in the iterable of a for loops is calculated before the loop.

Author: ```suggestion
-   #2039 : Ensure any expressions in the iterable of a for loop are calculated before the loop.
```
